configuration/rumqttd/rumqttd_5885.conf

```diff
@@ -0,0 +1,27 @@
+# Broker id. Used to identify local node of the replication mesh

```

Comment:
This configuration file must be moved along the tests in the `common/mqtt_client/tests` directory.

* It's used *only* for the tests.

Reply:
Yes, I had this plan as well. Thought of doing it as part of another PR. But ok to do it as part of this PR itself.

Reply:
On the topic of the config file, I'm a little bit resistant to use explicit config files as even though at the moment we have only a single file we may need to have some more later on and this would become a nightmare to maintain. 
I propose we use more of a hardcoded/functional configuration for the server, something along the lines:
```rust
fn get_rumqttd_config() -> librumqttd::Config {
    let router_config = rumqttlog::Config {
        id: 0,
        dir: "/tmp/rumqttd".into(),
        max_segment_size: 10240,
        max_segment_count: 10,
        max_connections: 10,
    };

    let connections_settings = ConnectionSettings {
        connection_timeout_ms: 1,
        max_client_id_len: 256,
        throttle_delay_ms: 0,
        max_payload_size: 10240,
        max_inflight_count: 200,
        max_inflight_size: 1024,
        username: None,
        password: None,
    };

    let server_config = ServerSettings {
        listen: SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 1884)),
        cert: None,
        next_connection_delay_ms: 1,
        connections: connections_settings,
    };

    let mut servers = HashMap::new();
    servers.insert("1".to_string(), server_config);

    let console_settings = ConsoleSettings {
        listen: SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 3030)),
    };

    librumqttd::Config {
        id: 0,
        router: router_config,
        servers,
        cluster: None,
        replicator: None,
        console: console_settings,
    }
}
```

This can be further extended as per need and we can add convenience methods as well such that it is easier to configure the server adhoc, which with the config file is not straightforward.

Reply:
Yes, such a function must be used instead of config files. One needs to add the port as a parameter.

Reply:
> On the topic of the config file, I'm a little bit resistant to use explicit config files as even though at the moment we have only a single file we may need to have some more later on and this would become a nightmare to maintain.
> I propose we use more of a hardcoded/functional configuration for the server, something along the lines:
> 
> ```rust
> fn get_rumqttd_config() -> librumqttd::Config {
>     let router_config = rumqttlog::Config {
>         id: 0,
>         dir: "/tmp/rumqttd".into(),
>         max_segment_size: 10240,
>         max_segment_count: 10,
>         max_connections: 10,
>     };
> 
>     let connections_settings = ConnectionSettings {
>         connection_timeout_ms: 1,
>         max_client_id_len: 256,
>         throttle_delay_ms: 0,
>         max_payload_size: 10240,
>         max_inflight_count: 200,
>         max_inflight_size: 1024,
>         username: None,
>         password: None,
>     };
> 
>     let server_config = ServerSettings {
>         listen: SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 1884)),
>         cert: None,
>         next_connection_delay_ms: 1,
>         connections: connections_settings,
>     };
> 
>     let mut servers = HashMap::new();
>     servers.insert("1".to_string(), server_config);
> 
>     let console_settings = ConsoleSettings {
>         listen: SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), 3030)),
>     };
> 
>     librumqttd::Config {
>         id: 0,
>         router: router_config,
>         servers,
>         cluster: None,
>         replicator: None,
>         console: console_settings,
>     }
> }
> ```
> 
> This can be further extended as per need and we can add convenience methods as well such that it is easier to configure the server adhoc, which with the config file is not straightforward.
I agree with this because we can avoid dependency on the config file.